core/types/src/macros.rs

```diff
@@ -178,4 +214,67 @@ mod test {
         let outer = Outer::default();
         assert_eq!(outer.0.field, [0u8; Outer::SIZE]);
     }
+
+    #[test]
+    fn newtype_byte_array_display() {
+        let outer = Outer::from([
+            0xABu8, 0x00, 0xcd, 0x12, 0xfe, 0x1, 0x2, 0x3, 0x4, 0x5, 0x6, 0x7, 0x8, 0x9, 0x0a,
+            0x0B, 0x0C, 0x0D, 0x0E, 0x0F, 0x10, 0x11, 0x12, 0x13,
+        ]);
+        assert_eq!(
+            outer.to_string(),
+            "0xAB00CD12FE0102030405060708090A0B0C0D0E0F10111213"
```

Comment:
Note this implementation is upper hex, I don't have a preference. we could leverage alternate `#` for upper vs lower, but I don't think many people will think to look there to toggle case.